flexmeasures/api/common/utils/deprecation_utils.py

```diff
@@ -1,12 +1,39 @@
 from __future__ import annotations
 
-from flask import current_app, request, Blueprint, Response, after_this_request
+from flask import abort, current_app, request, Blueprint, Response, after_this_request
 from flask_security.core import current_user
 import pandas as pd
 
 from flexmeasures.utils.time_utils import to_http_time
 
 
+def sunset_blueprint(
+    blueprint,
+    api_version_sunset: str,
+    sunset_info_link: str,
+    api_version_upgrade_to: str = "3.0",
+):
+    """Sunsets every route on a blueprint by returning 410 (Gone) responses.
+
+    Such errors will be logged by utils.error_utils.error_handling_router.
+    """
+
+    def let_host_switch_to_returning_410():
+
+        # Override with custom info link, if set by host
+        _sunset_info_link = current_app.config.get(
+            "FLEXMEASURES_API_1_AND_2_SUNSET_LINK", sunset_info_link
```

Comment:
This does not work well. I haven't got the config setting, and my message for API 2.0 reads "API version 2.0 has been sunset. Please upgrade to API version 3.0. See None for more information."

It should use the link from the blueprint. Apparently, the fallback is not used.

Reply:
Good catch!